Tests/GraphQLTransportWSTests/GraphQLTransportWSTests.swift

```diff
@@ -189,21 +191,167 @@ class GraphqlTransportWSTests: XCTestCase {
         client.onMessage { message, _ in
             messages.append(message)
         }
-        
+
         client.sendConnectionInit(
             payload: TokenInitPayload(
                 authToken: ""
             )
         )
-        
+
         wait(for: [completeExpectation], timeout: 2)
         XCTAssertEqual(
             messages.count,
             5, // 1 connection_ack, 3 next, 1 complete
             "Messages: \(messages.description)"
         )
     }
-    
+
+    /// Tests a streaming conversation where the client sends data to the server
+    func testStreamingClientServer() throws {
+        var messages = [String]()
+        let completeExpectation = XCTestExpectation()
+
+        let client = Client<TokenInitPayload>(messenger: clientMessenger)
+        client.onConnectionAck { _, client in
+            client.addObservable(observable: testObservable(loop: self.eventLoop))
+        }
+
+        client.onError { _, _ in
+            completeExpectation.fulfill()
+        }
+
+        server.onMessage { message in
+            messages.append(message)
+        }
+        server.onNext { _, _ in
+            completeExpectation.fulfill()
+        }
+
+        client.sendConnectionInit(
+            payload: TokenInitPayload(
+                authToken: ""
+            )
+        )
+
+        wait(for: [completeExpectation], timeout: 2)
+        XCTAssertEqual(
+            messages.count,
+            2, // 1 connection_init, 1 next
+            "Messages: \(messages.description)"
+        )
+    }
+
+    /// Test a conversation where both ends are sending data at the same time
+    func testBiderectionalStreaming() throws {
+        let id = UUID().uuidString
+
+        var clientMessages = [String]()
+        var serverMessages = [String]()
+
+        let clientCompleteExpectation = XCTestExpectation()
+        let serverCompleteExpectation = XCTestExpectation()
+
+        var dataIndex = 1
+        let dataIndexMax = 3
+
+        let client = Client<TokenInitPayload>(messenger: clientMessenger)
+        client.onConnectionAck { _, client in
+            client.sendSubscribe(
+                payload: GraphQLRequest(
+                    query: """
+                    subscription {
+                        hello
+                    }
+                    """
+                ),
+                id: id
+            )
+
+            // Short sleep to allow for server to register subscription
+            usleep(3000)
+
+            pubsub.onNext("hello \(dataIndex)")
+
+            // Once the subscription has begun send a `next` frame from the client as well
+            client.addObservable(observable: testObservable(loop: self.eventLoop))
```

Comment:
How are we triggering the `next` frame back to the server? It looks like this just adds the observable. Maybe it's just the function signature that's confusing me.

nit: Consider adding this message back into the `client.onNext`, and then putting the `pubsub.onNext` into the `server.onNext` for a more complete conversation. For example:

```
Server -> Client: connection_ack
Client -> Server: subscribe
Server -> Client: next (hello)
Client -> Server: next
Server -> Client: next
...
Client -> Server: complete
```

Reply:
updated name to `addObservableSubscription` to hopefully clarify

Whatever observable you register by calling this function turns around and subscribes, converting `observable.onNext` callbacks into `GraphQLTransportS.NextResponse`s and sending those through the messenger.